### ISSUE TITLE
Micro frontends approach

### ISSUE THREAD

**Issue by cv65kr** [uber/baseweb#4023]
Hey 

I have couple questions about approach needed to using UI and micro frontends. 

1. Does exists possibility to add prefix to class names generated in scope of Uber UI library? I mean for default compontents. For example if I add or modify something, it should not have prefix.

2. For example I have two micro frontend services and service which merge both micro services. How avoid generating duplicates of class names in result service?

3. Does exists possiblity to compile all default class names cases to css file. I was thining about global assets service to avoid duplicates from point 2.

Questions may looks weird but I am looking to some way to avoid duplicating of class names in result service.

Maybe someone have experience with micro frontends and Uber UI and want to share your knowledge. Problems also are welcome.

**Comment by chasestarr** [comment]
I believe you can add a class name prefix when instantiating the styletron client. Docs here https://www.styletron.org/api-reference#options

**Comment by cv65kr** [comment]
Yes it's possible but problem is quite deeper. If you add engine to `StyletronProvider` like on example (https://baseweb.design/guides/bidirectionality/) eg. `new Styletron('my_prefix_')` it will be works. Main problem is impact not only for vendor base ui components but also for those create in my app.

Example:
```
const engine = new Styletron('my_prefix_');

<StyletronProvider value={engine}>
      <BaseProvider theme={{...LightTheme, direction: 'rtl'}}>
        <StatefulInput /><MyComponent>
      </BaseProvider>
    </StyletronProvider>
```

Result:
```
<input class="my_prefix_aa my_prefix_bb ..." />
<div id="my-component" class="my_prefix_cc my_prefix_dd"></div>
```

Expected result
```
<input class="my_prefix_aa my_prefix_bb ..." />
<div id="my-component" class="cc dd"></div>
```
